doc/spec.md

```diff
@@ -1,222 +1,174 @@
-manager. In version 1.0, the only supported package manager is
-opam.  The user only needs to have opam installed but not setup,
-since this tool takes care of initialising it.
+It should also make it easy for you to work on your dependencies, modify them, open PRs upstream
+or push your changes to forks to be able to reuse them while making sure none of this breaks your
+project or any of your other dependencies.
 
-Finally, the duniverse tool is tightly integrated with git. In
-particular, the workspaces it creates and manages will always be
-versioned controled by git. This ensures that users can access the
-history of the workspace as well as of individual projects inside it.
+`duniverse` relies on a `duniverse.sxp` file, at the root of the project to operate. It contains
+all the information needed to gather and inspect all of your transitive dependencies.
+For each package it know its name, the upstream git repository URL, the specific git reference
+for the version you're using as well as the eventual fork repository URL you're using instead of
+the upstream one.
+The `duniverse.sxp` file should be part of your project repository as it specifies how to locally
+pull your dependencies to be able to build your project.
+It is in a human readable sexp format and you should be able to edit it by hand in some cases where
+the `duniverse` tool doesn't cover your specific use case.
 
-Specification
--------------
+`duniverse` will pull your dependencies in a `duniverse/` directory at the root of your project.
+It will clone the sources using `git` and check them out to the right tag, branch or commit. 
+The `duniverse/` directory contains a subdir for each of your transitive dependencies, named after
+the opam package they correspond to or using the name you provided when adding a non-opam
+dependency.
+This directory will be setup so that running `dune build` or `dune runtest` from the root of your
+project will only build your public targets and run your tests but not the ones in `duniverse/`
+unless you explictly depend on them.
+Each subdir in `duniverse/` is a regular git project, meaning that you can easily `cd` into one
+and start working on it with your regular git workflow. The `duniverse` tool provides helpers to
+manage those changes and keep track of them in the `duniverse.sxp` file.
+
+Because `duniverse` only supports packages that are built with `dune`, it will complain if a
+dependency cannot be resolved to a version that is built with `dune`. Hopefully, to work around
+that it will also look into the [`dune-universe/opam-overlays`]() opam repository for dunified
+forks of package which haven't been ported upstream yet. When a dependency from the `opam-overlays`
+repo is picked, it will consider it's git URL as the upstream one and not the actual upstream
+package repo URL.
+
+### System constraints
+
+`duniverse` requires that your system has OCaml and `dune` installed, in the right versions.
```

Comment:
We could also vendor dune -- since 1.7, it supports being in the same directory. github.com/avsm/platform has a bootstrap script that builds dune first and uses that to build the duniverse, implying (like go) that the only thing needed to build an executable is ocaml itself.  And if we really wanted to, we could also bootstrap OCaml of course :-)